# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/SharedContext.cpp
# Commit: f505b8fb8f8b
# Full Hash: f505b8fb8f8b7a8784be192ce6a36dc2278095cb
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 8: Use TaggedParserAtomIndex in ScriptStencil.functionAtom. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96915
# ==============================================================================

diff -r 3749d6d2325d -r f505b8fb8f8b js/src/frontend/SharedContext.cpp
--- a/js/src/frontend/SharedContext.cpp	Sat Nov 14 00:03:00 2020 +0000
+++ b/js/src/frontend/SharedContext.cpp	Sat Nov 14 00:03:07 2020 +0000
@@ -431,8 +431,8 @@
 
   if (atom_) {
     atom_->markUsedByStencil();
+    script.functionAtom = atom_->toIndex();
   }
-  script.functionAtom = atom_;
   script.functionFlags = flags_;
   script.nargs = nargs_;
   script.lazyFunctionEnclosingScopeIndex_ = enclosingScopeIndex_;
@@ -467,8 +467,8 @@
   ScriptStencil& script = functionStencil();
   if (atom_) {
     atom_->markUsedByStencil();
+    script.functionAtom = atom_->toIndex();
   }
-  script.functionAtom = atom_;
   script.functionFlags = flags_;
 }
 